android_vs_ios.md

```diff
@@ -0,0 +1,175 @@
+Historically, layouts are generally built by writing xml instead of using the interface builder, partly due to Android Studio's interface builder being terrible/inadequate. The interface builder has improved quite a lot recently with the introduction of ConstraintLayout, so it has more usage nowadays. However, building the layout via by xml might still be more preferable to a lot of people since it offers more control, and Android Studio can help autocomplete many of the fields.
+
+## Libraries
+[Gradle](https://developer.android.com/studio/build/), the official build system for Android, includes the ability to manage libraries and dependencies.\
+iOS projects generally use [CocoaPods](https://cocoapods.org/) for managing libraries. [Swift Package Manager](https://swift.org/package-manager/) is the official dependency manager for Swift that was released somewhat recently, although Cocoapods is still more widely used.
+
+Due to slow/nonexistent updates from Android device manufacturers, many Android devices are stuck on older version of Android. This makes it very difficult to develop for as certain features might only be available on newer version Android and certain method calls behave differently on different OS versions. Google provides [Android Support Library](https://developer.android.com/topic/libraries/support-library/) that backports certain new features to older OS version, provides consistent behavior across different OS version, and generally abstracts away a lot of boilerplate when dealing with backwards compatibility. **It is highly recommended to use the support library calls/classes instead of platform version whenever possible.** Additionally, Google recently introduces [Architecture Components](https://developer.android.com/topic/libraries/architecture/) that tries solve some common headaches in Android development such as dealing with lifecycle and database.\
+There are no equivalent of these on iOS.
+
+## Testing
+Android runs [unit tests](https://developer.android.com/training/testing/unit-testing/local-unit-tests) on the computer’s JVM, without Android device or emulator. All Android calls are stubbed out and often just throws an exception when invoked in tests. This means testable classes (Presenter, ViewModels) should not directly include any Android classes/calls. There is a library called [Robolectric](http://robolectric.org/) which tries to reimplement all the Android calls in JVM, but it should be used sparingly since it’s slower and relies too much on “fake” Android code.\
```

Comment:
As of the next version (4.0) Robolectric is part of Jetpack, so we should expect people to be using it more. And as I understand it they're working with the Android team to move away from "fake" implementations as part of this update.

Reply:
we can update this once it's more official